### ISSUE TITLE
Enhance Etcd spec to include snapshot compaction spec

### ISSUE THREAD

**Issue by shreyas-s-rao** [gardener/etcd-druid#708]

**Comment by unmarshall** [comment]
@shreyas-s-rao  we already have [CompactionResources](https://github.com/gardener/etcd-druid/blob/794de68b2f41f8484de5e94b91ac7ec18c895acd/api/v1alpha1/etcd.go#L151) as part of the `Backup` spec. This is also used in the compaction reconciler. See [here](https://github.com/gardener/etcd-druid/blob/794de68b2f41f8484de5e94b91ac7ec18c895acd/internal/controller/compaction/reconciler.go#L350-L352)

**Comment by shreyas-s-rao** [comment]
@unmarshall yes it should. It's up to the consumer to utilise this spec field to set resource requests/limits for the compaction job pods. @anveshreddy18 can you confirm if you added this to g/g when gardenlet creates the `Etcd` resources?

**Comment by shreyas-s-rao** [comment]
@anveshreddy18 But it makes sense to utilise this spec field for g/g, rather than hardcoding the values in the reconciler. Can you please make this change?

**Comment by ashwani2k** [comment]
[like]  Kumar, Ashwani reacted to your message:
________________________________
